tools/bazar/fields/MapField.php

```diff
@@ -230,43 +244,47 @@ function geocodedmarkerRefresh( point )
         $GLOBALS['wiki']->AddJavascript($initMapScript.$geoCodingScript);
 
         return $this->render("@bazar/inputs/map.twig", [
-            'value' => $this->getValue($entry),
-            'latitude' => $latitude,
-            'longitude' => $longitude
+            'latitude' => $value[$this->getLatitudeField()],
+            'longitude' => $value[$this->getLongitudeField()]
         ]);
     }
 
     public function formatValuesBeforeSave($entry)
     {
         if (!$this->canEdit($entry)) {
             // retrieve value from value because redefined with right value
-            $value = $this->getValue($entry);
-            $values = (empty($value)) ? null : explode('|', $value);
-            if (empty($values[0]) || empty($values[1])) {
+            $values = $this->getValue($entry);
+            if (empty($values[$this->getLatitudeField()]) || empty($values[$this->getLongitudeField()])) {
```

Comment:
```suggestion
            if (empty($values)) {
```

Est-ce que ça ne serait pas suffisant vu que getValue fait déjà le test ?